### ISSUE TITLE
[Paper]: Location is Key: Leveraging LLM for Functional Bug Localization in Verilog Design

### ISSUE THREAD

**Issue by llTurtle22ll** [juliodosreis/awesome-agenticsystems#10]
### Link to the paper

https://dl.acm.org/doi/10.1109/DAC63849.2025.11133280

### BibTeX (optional)

@inproceedings{10.1109/DAC63849.2025.11133280,
author = {Yao, Bingkun and Wang, Ning and Zhou, Jie and Wang, Xi and Gao, Hong and Jiang, Zhe and Guan, Nan},
title = {Location is Key: Leveraging LLM for Functional Bug Localization in Verilog Design},
year = {2025},
isbn = {9798331503048},
publisher = {IEEE Press},
url = {https://doi.org/10.1109/DAC63849.2025.11133280},
doi = {10.1109/DAC63849.2025.11133280},
abstract = {In Verilog code design, identifying and locating functional bugs is an important yet challenging task. Existing automatic bug localization methods have limited capabilities; they only suggest a set of potential buggy lines rather than precisely identifying the bug. Moreover, they depend on verification tools like testbenches and reference models, which require expert input and are time-consuming to develop. This paper introduces LiK (Location is Key), an open-source Large Language Model (LLM) to precisely locate functional bugs in Verilog code without the need for expert-written verification tools. LiK is developed from the open-source coding LLM Deepseek-Coder-Lite-Base-16B through a three-step training process: continuous pre-training to enhance foundational knowledge, supervised fine-tuning to learn how to output localization results, and reinforcement learning to reduce output errors. Experiment results demonstrate that LiK achieves superior functional bug localization accuracy, outperforming both the SOTA traditional method Strider, and SOTA closed-source LLMs like GPT-o1-preview and Claude-3.5-Sonnet. Moreover, integrating LiK into the SOTA LLM-based Verilog debugging tool significantly boosts its functional bug fixing success rate from 76.47\% to 90.54\%. This underscores LiK's potential to enhance the performance of end-to-end automatic Verilog debugging tools.},
booktitle = {Proceedings of the 62nd Annual ACM/IEEE Design Automation Conference},
articleno = {394},
numpages = {7},
location = {San Francisco, California, United States},
series = {DAC '25}
}

### Area

tools-context — How the agent acts on the world and where it gets context from.

### Scale

single-agent — One agent acting on its own.

### Type

method — Proposes a new technique.

**Comment by llTurtle22ll** [comment]
**Submited by:** Ângelo Malaguti e Edgar Galvão - 165429 e 300644 - MO810
